resource/js/helpers.js

```diff
@@ -161,6 +172,61 @@ app.directive("customDropdown", function() {
     }
 })
 
+// For now, key can only be one of a preset of values (dropdown)
+// & the value field only accepts numbers
+app.directive("keyValueList", function($timeout) {
+    return {
+        scope: {
+            keyOptions: '=',
+            map: '=',
+            keyLabel: '@',
+            keyItemLabel: '@',
+            valueLabel: '@',
+            valueRange: '@',
+            form: "="
+        },
+        restrict: 'A',
+        templateUrl:'/plugins/stress-test-center/resource/templates/key-value-list.html',
+        link: function(scope) {
+            scope.keys = [null];
+            [ scope.valueMin, scope.valueMax ] = scope.$eval(scope.valueRange) || [null, null];
+            const VALIDITY = "key-value-list-valid";
+            const DEFAULT_VALUE = .5;
```

Comment:
This should perhaps be a parameter since this is a generic key value list?

Reply:
Done in 653a005